src/components/app/KeyboardShortcut.js

```diff
@@ -0,0 +1,270 @@
+    if (focusAfterClosed) {
+      requestAnimationFrame(() => {
+        // Restore focus, but not during a React setState call, otherwise this triggers
+        // a React warning:
+        // "unstable_flushDiscreteUpdates: Cannot flush updates when React is already rendering."
+        focusAfterClosed.focus();
+      });
+    }
+
+    return { isOpen: false, focusAfterClosed: null };
+  };
+
+  _handleCloseClick = () => {
+    this.setState(this._close);
+  };
+
+  _handleKeyPress = (event: KeyboardEvent) => {
+    const target = coerce<EventTarget, HTMLElement>(event.target);
+    switch (event.key) {
+      case '?': {
+        if (
+          target.tagName === 'INPUT' ||
+          target.tagName === 'TEXTAREA' ||
+          target.tagName === 'SELECT'
+        ) {
+          // Ignore this from input-like things.
+          return;
+        }
+        // Toggle the state.
+        this.setState(state =>
+          state.isOpen ? this._close(state) : this._open(state)
+        );
+        break;
+      }
+      case 'Escape': {
+        // Unconditionally run close on escape, which is a noop if it's not open.
+        this.setState(this._close);
+        break;
+      }
+      default:
+      // Do nothing.
+    }
+  };
+
+  componentWillUnmount() {
+    window.removeEventListener('keydown', this._handleKeyPress);
+    this._untrapFocus();
+  }
+
+  _trapFocus() {
+    document.addEventListener('focus', this._trapFocusHandler, true);
+  }
+
+  _untrapFocus() {
+    document.removeEventListener('focus', this._trapFocusHandler, true);
+  }
+
+  // This is inspired by:
+  // https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html
+  _trapFocusHandler = (event: FocusEvent) => {
+    const div = this._focusArea.current;
+    if (!div) {
+      return;
+    }
+    if (!div.contains(coerce<EventTarget, Node>(event.target))) {
+      // TODO - This does not handle shift-tabbing going to the last focusable
+      // element in the list.
+      div.focus();
+    }
+  };
+
+  renderShortcuts() {
+    if (!this.state.isOpen) {
+      return null;
+    }
+
+    return (
+      <>
+        <div className="appKeyboardShortcutsHeader">
+          <div
+            className="appKeyboardShortcutsHeaderTitle"
+            /* aria-labelledby id */
+            id="AppKeyboardShortcutsHeaderTitle"
+          >
+            Keyboard shortcuts
+          </div>
+          <button
+            type="button"
+            className="appKeyboardShortcutsHeaderClose"
+            onClick={this._handleCloseClick}
+          >
+            Close
+          </button>
+        </div>
+        <div className="appKeyboardShortcutsContent">
+          <div className="appKeyboardShortcutsColumn">
+            <h2>Call Tree</h2>
+            <Shortcut label="Close call node" shortcut="ArrowLeft" />
+            <Shortcut label="Open call node" shortcut="ArrowRight" />
+            <Shortcut label="Open all child call nodes" shortcut="*" />
```

Comment:
Oh wow, I didn't know about this actually. I was always doing the "option+click" on the triangle thingy. I wonder if we can also put this here now. But it's not exactly a shortcut 🤔

Reply:
I'd be fine adding it, but I'm not really sure how to actually document "click on the triangle thingy". Also, I can't seem to reproduce this feature? I'll defer this for now.